### ISSUE TITLE
Suggestion: Add keybinding to execute a preview command

### ISSUE THREAD

**Issue by lilyball** [junegunn/fzf#1638]
- Category
    - [X] fzf binary
    - [ ] fzf-tmux script
    - [X] Key bindings
    - [ ] Completion
    - [ ] Vim
    - [ ] Neovim
    - [ ] Etc.

It would be neat if fzf had key binding action to run a command in the preview window, similar to `execute`.

This was inspired by https://paweldu.dev/posts/fzf-live-repl/, which uses `fzf` as a REPL for quickly running various inputs through commands, but has the downside of running it on every keypress which may be dangerous depending on the command being tested. If I could bind a key (such as enter) to something like `preview(cat someFile | awk {q})` then that would give me control over when the command runs.

An alternative would be something like a flag `--preview-no-auto-refresh` in conjunction with a bind action `refresh-preview` so I could then do `--preview="cat someFile | awk {q}" --preview-no-auto-refresh --bind "enter:refresh-preview"`, but that seems rather awkward, and the only benefit I see to this approach is reducing confusion over what happens if I specify both `--preview` and `--bind "enter:preview(…)"`.

**Comment by sholland1** [comment]
I was able to get the preview to refresh by toggling the preview twice: `--bind='enter:toggle-preview+toggle-preview'`

**Comment by lilyball** [comment]
A benefit of being able to do something like `--bind "enter:preview(…)"` is being able to bind different keys to do different previews. For example, if I wanted to filter by json files, I could have the normal preview just show the file and then bind a key to pipe the file through `jq` and back into the preview window, in order to prettify it.

**Comment by lilyball** [comment]
Actually, looking at the original issue again, the suggested resolution doesn't address the problem. The problem isn't that I want to refresh preview, it's that I want to *avoid* refreshing preview on every keypress even though the preview command includes the query string. Avoiding refreshing it on every keypress does require being able to then trigger a refresh, but without the ability to suppress the auto-refresh it doesn't help.

**Comment by sholland1** [comment]
Would the `--phony` flag help?

**Comment by lilyball** [comment]
No, that flag changes nothing at all about the example shown in https://paweldu.dev/posts/fzf-live-repl/.

**Comment by lilyball** [comment]
I suppose a hacky way to accomplish this is to skip the preview and instead do something like `--bind "enter:execute(cat file | awk {q} | less)"`, but having it switch away from fzf to show the output would be annoying.

Ignoring the fzf-live-repl link for a moment, I think it might be more useful to just focus on the use-case of wanting to be able to manually transform the preview. For example, I want to be able to do something like `find . -name '*.json' | fzf --preview cat --bind 'ctrl-f:preview(jq -C <{})'`. This way it would just show the file contents by default, but ctrl-f would update the preview to show the formatted version using `jq`.

**Comment by junegunn** [comment]
Have you actually tested the new code? It seems to me that it addresses all your concerns.

**Comment by lilyball** [comment]
I'm embarrassed to admit I didn't realize this was actually resolved with a commit. I came back to this ticket because I saw sholland1's comment, and I saw that the ticket was closed, but I didn't look closely at the reason for it being closed.

Now that I'm aware, I haven't tested the new code as it's not in a published release, but the manpage for the new code sounds like it does precisely what I want, thanks.
